test/C/src/modal_models/ModalStateResetAuto.lf

```diff
@@ -0,0 +1,98 @@
+/*
+ * Modal Reactor Test.
+ * Tests reset of state variables in modes.
+ */
+target C {
+    fast: false,
+    timeout: 4 sec
+};
+
+import TraceTesting from "util/TraceTesting.lf"
+
+reactor Modal {
+    input next:bool;
+    
+    output mode_switch:int
+    output count0:int;
+    output count1:int;
+    output count2:int;
+    
+    state counter0:int(0);
+    
+    reaction(next) -> count0 {=
+        printf("Counter0: %d\n", self->counter0);
+        lf_set(count0, self->counter0++);
+    =}
+    
+    initial mode One {
+        state counter1:int(0);
```

Comment:
It looks like the state variable isn't rendered quite correctly in the diagram:
![image](https://user-images.githubusercontent.com/19938940/173119911-40db4248-d00a-4c1a-85be-d855e5785d9d.png)